R/errorChecking.R

```diff
@@ -1,19 +1,10 @@
+#' Check response of API call, produce warning on error
+#' 
+#' @param response httr response object from of API call
+
 checkHttpErrors <- function(response) {
 
   if(response["status_code"] != 200) {
       stop(paste0(httr::content(response), " \nHTTP error details = ", httr::http_status(response)$message))
```

Comment:
How do we handle this in a shiny dashboard for example?

Let's say the dashboard calls the aggregate-data function which calls this function. If there is an error and we call the function `stop`, how happens in the dashboard? Do we display the error message? Is there a way to `catch` the error and display a friendly message?

Reply:
Yes, we catch the error sometimes, though this isn't implemented consistently. [Example here](https://github.com/lanalabs/purchase-to-pay/blob/551f03cd3804dbff109eda462dabf41a5619a5ac/Shiny-Dashboard/pages/discount.Rmd#L19)

I agree it would be nice to try/catch all elements. We should discuss and maybe add a card.